app/utils/temperature_api.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+import httpx
+from fastapi import HTTPException
+
+WEATHER_API_KEY = os.getenv("WEATHER_API_KEY")

```

Comment:
Ensure that `WEATHER_API_KEY` is set in the environment variables before running the application. If it's not set, the API request will fail. Consider adding a check to raise an informative error if the key is missing.